utils/UniswapPoolsHelpers.ts

```diff
@@ -1,52 +1,40 @@
-import type { SignerWithAddress } from "hardhat-deploy-ethers/dist/src/signers";
 import type { HardhatRuntimeEnvironment } from "hardhat/types";
 import type { IERC20 } from "../typechain/IERC20";
-import { getBdx, getWeth, getWbtc, getUniswapPairOracle, getBot, getAllBDStables, formatAddress } from "./DeployedContractsHelpers";
+import { getBdx, getWeth, getWbtc, getUniswapPairOracle, getAllBDStables, formatAddress } from "./DeployedContractsHelpers";
 import * as constants from "../utils/Constants";
+import { Recorder } from "./Recorder/Recorder";
+import { toRc } from "./Recorder/RecordableContract";
 import { getListOfSupportedLiquidityPools } from "../utils/Constants";
-import { printAndWaitOnTransaction } from "./DeploymentHelpers";
-
-export async function updateUniswapPairsOracles(hre: HardhatRuntimeEnvironment, signer: SignerWithAddress | null = null) {
-  console.log("Starting tp update the Uniswap oracles");
 
+export async function recordUpdateUniswapPairsOracles(hre: HardhatRuntimeEnvironment, recorder: Recorder) {
   const pools = await getPools(hre);
+  const promises = pools.map(pool => recordUpdateOracle(hre, pool[0].name, pool[1].name, recorder));
 
-  for (const pool of pools) {
-    await updateOracle(hre, pool[0].name, pool[1].name, signer);
-  }
+  await Promise.allSettled(promises);
+}
 
-  console.log("Finished updating the Uniswap oracles");
+export async function recordUpdateOracle(hre: HardhatRuntimeEnvironment, symbol0: string, symbol1: string, recorder: Recorder) {
+  const oracle = toRc(await getUniswapPairOracle(hre, symbol0, symbol1), recorder);
+  await oracle.record.updateOracle();
```

Comment:
We had this on purpose:
```js
// Do not await on the function so we could run multiple transactions like that in parallel
  printAndWaitOnTransaction(await oracle.connect(updater).updateOracle()).catch(error => console.log(`Error while updating ${oracleName}`, error));
```

for few reasons:
1. To print an error in a failed transaction
2. To print the transaction hash when executing it (using the `printAndWaitOnTransaction` function)
3. To make sure everything is being executed in parallel and not wait for each oracle, while still being able to execute the transactions. Before the change in `updateUniswapPairsOracles`, it had the Promise.allSettled approach you brought back here. The problem there was that you can't execute it in parallel this way as everything gets the same nonce and doesn't work.. This is why it was changed to a for loop without `await Promise.allSettled(promises);`. I believe this code now doesn't really work in parallel like it used to be and we better bring it back to what it was in this function and in `recordUpdateUniswapPairsOracles` (with your `toRc` code of course)

Reply:
@kazazor firstly, now it's the recorder's strategy job to handle transactions.

If we use immediate execution strategy it will wait for every transaction execution before it moves to the next one: https://github.com/BlindexDAO/BlinDEX/blob/lago-628-hardhat-tasks-multisig/utils/Recorder/strategies/ImmediateExecutionStrategy.ts#L28

If we use timelock strategy transactions will be batched and executed sequentially as internal transactions

So I think we're good here. WDYT?

Reply:
Let's just add the `console.log(`Transaction Hash: ${transaction.hash}`);` to the strategies